lib/Cache.php

```diff
@@ -2,415 +2,431 @@
-            }
-        }
-        $index_path = $this->get_path('index');
-        $this->write_cache($index_path, $this->indexes);
+    $index_path = $this->get_path("index");
+    $this->write_cache($index_path, $this->indexes);
+  }
+
+  /**
+   * Get cache content
+   *
+   * @return string
+   */
+  public function get_cache(): string
+  {
+    $args = func_get_args();
+    $cache_path = call_user_func_array([$this, "get_path"], $args);
+
+    if (is_file($cache_path)) {
+      return file_get_contents($cache_path);
     }
-
-    /**
-     * Get cache content
-     *
-     * @return string
-     */
-    public function get_cache()
-    {
-        $args = func_get_args();
-        $cache_path = call_user_func_array(array($this, 'get_path'), $args);
-
-        if (is_file($cache_path)) {
-            return file_get_contents($cache_path);
-        }
-        return null;
+    return null;
+  }
+
+  /**
+   * Write to cache atomically
+   *
+   * @param  string $cache_path
+   * @param  mixed $content
+   * @return int
+   */
+  public function write_cache($cache_path, $content): int
+  {
+    $cache_content = json_encode($content);
+    $cache_size = strlen($cache_content);
+
+    $temp = tempnam($this->params["path"], "temp");
+    if (!($f = @fopen($temp, "wb"))) {
+      $temp = $this->params["path"] . "/" . uniqid("temp");
+      if (!($f = @fopen($temp, "wb"))) {
+        throw new WriteException("Unable to write temporary file " . $temp);
+      }
     }
 
-    /**
-     * Write to cache atomically
-     *
-     * @param  string $cache_path
-     * @param  mixed $content
-     * @return int
-     */
-    public function write_cache($cache_path, $content)
-    {
-        $cache_content = json_encode($content);
-        $cache_size = strlen($cache_content);
-
-        $temp = tempnam($this->params['path'], 'temp');
-        if (!($f = @fopen($temp, 'wb'))) {
-            $temp = $this->params['path'].'/'.uniqid('temp');
-            if (!($f = @fopen($temp, 'wb'))) {
-                throw new WriteException('Unable to write temporary file '.$temp);
-            }
-        }
-
-        fwrite($f, $cache_content);
-        fclose($f);
-
-        if (!rename($temp, $cache_path)) {
-            unlink($cache_path);
-            rename($temp, $cache_path);
-        }
+    fwrite($f, $cache_content);
+    fclose($f);
 
-        chmod($cache_path, $this->params['write_perms']);
-
-        return $cache_size;
+    if (!rename($temp, $cache_path)) {
+      unlink($cache_path);
+      rename($temp, $cache_path);
     }
 
-    /**
-     * Clear a cache path
-     *
-     * @param  string $cache_path
-     * @return void
-     */
-    public function clear_cache($cache_path)
-    {
-        @unlink($cache_path);
+    chmod($cache_path, $this->params["write_perms"]);
+
+    return $cache_size;
+  }
+
+  /**
+   * Clear a cache path
+   *
+   * @param  string $cache_path
+   * @return void
+   */
+  public function clear_cache($cache_path): void
```

Comment:
required php version is 7.1

it seems we need to display this in composer.json

Reply:
Thanks, resolved that!